### ISSUE TITLE
Path problem 

### ISSUE THREAD

**Issue by danama** [onedr0p/telegram-sonarr-bot#52]
Hello,

It seems that I have some problems in my configuration. Probably "path" related somewhere.

**Sysinfo** : 
- Debian Jessie
- node v4.5.0
- nodejs v4.5.0
- Sonarr on localhost with mono (http:/localhost:8989)

**Working**: 
1.  Each command are well received by telegram-sonarr-bot from telegram. (logs)
2. Both notification and request reply correctly in the same channel on telegram (no id issue)
3. /upcoming
4. /library

**Not working**:
1. If I do a query, everything looks fine but when the series is added nothing happened in sonarr. The series is not present in the library and in sonarr.log I have this : 

`{"level":"error","message":"err:Error: ENOENT: no such file or directory, stat 'Unknown Path'","timestamp":"2016-08-30T16:00:39.790Z"}`

2. The notifications received doesn't have any information except the name of the show when I manually grab a show in sonarr : 
```

Episode Imported
Battlestar Galactica (2003) - Unknown SeasonxUnknown Episode
Air Date: Unknown Air Dates
Quality: Unknown Quality
Size: 0 MB
```
and in the logs, I have: 

`[Error] node: 2016-08-30T16:00:39.789Z - #033[31merror#033[39m: err:Error: ENOENT: no such file or directory, stat 'Unknown Path'`

Any idea of where is my problem ?

Thanks

**Comment by onedr0p** [comment]
Do you mind posting the complete log for the process of adding that show?

**Comment by danama** [comment]
Sure,

This one is from sonarr.log

```
{"level":"info","message":"user: Dan undefined, message: sent '/query' command","timestamp":"2016-09-01T12:06:27.652Z"}
{"level":"info","message":"user: Dan undefined, message: requested to search for series \"fringe\"","timestamp":"2016-09-01T12:06:27.808Z"}
{"level":"info","message":"#2 user: Dan undefined, message: found the following series Fringe - 2008","timestamp":"2016-09-01T12:06:27.816Z"}
{"level":"info","message":"user: 208282568, message: confirm the series Fringe - 2008","timestamp":"2016-09-01T12:06:36.163Z"}
{"level":"info","message":"user: Dan undefined, message: looking for existing series","timestamp":"2016-09-01T12:06:36.540Z"}
{"level":"info","message":"user: Dan undefined, message: conform correct show: Fringe - 2008","timestamp":"2016-09-01T12:06:36.657Z"}
{"level":"info","message":"user: 208282568, message: choose the series Yes","timestamp":"2016-09-01T12:06:45.149Z"}
{"level":"info","message":"user: Dan undefined, message: requested to get profile list","timestamp":"2016-09-01T12:06:45.159Z"}
{"level":"info","message":"#3 user: Dan undefined, message: found the following profiles Any,SD,HD-720p,HD-1080p,HD - All,WEB-DL 1080P,WEB-DL 720P,Bluray 1080P,Bluray 720p","timestamp":"2016-09-01T12:06:45.160Z"}
{"level":"info","message":"user: 208282568, message: choose the profile \"WEB-DL 1080P\"","timestamp":"2016-09-01T12:06:52.517Z"}
{"level":"info","message":"user: Dan undefined, message: requested to get monitor list","timestamp":"2016-09-01T12:06:52.518Z"}
{"level":"info","message":"user: Dan undefined, message: found the following monitor types future,all,none,latest,first","timestamp":"2016-09-01T12:06:52.518Z"}
{"level":"info","message":"user: 208282568, message: choose the type \"none\"","timestamp":"2016-09-01T12:06:55.116Z"}
{"level":"info","message":"user: Dan undefined, message: requested to get series types list","timestamp":"2016-09-01T12:06:55.117Z"}
{"level":"info","message":"user: Dan undefined, message: found the following series types standard,airs daily,anime","timestamp":"2016-09-01T12:06:55.117Z"}
{"level":"info","message":"user: 208282568, message: choose the folder \"standard\"","timestamp":"2016-09-01T12:06:57.192Z"}
{"level":"info","message":"user: Dan undefined, message: requested to get folder list","timestamp":"2016-09-01T12:06:58.659Z"}
{"level":"info","message":"user: Dan undefined, message: found the following folders /mnt/ZFSPool/TvShows/","timestamp":"2016-09-01T12:06:58.660Z"}
{"level":"info","message":"user: 208282568, message: choose the season folder \"/mnt/ZFSPool/TvShows/\"","timestamp":"2016-09-01T12:07:03.296Z"}
{"level":"info","message":"user: Dan undefined, message: requested to get season folders list","timestamp":"2016-09-01T12:07:03.297Z"}
{"level":"info","message":"user: Dan undefined, message: found the following seasons folder types yes,no","timestamp":"2016-09-01T12:07:03.297Z"}
{"level":"info","message":"user: Dan undefined, message: adding series \"Fringe\" with options {\"tvdbId\":82066,\"title\":\"Fringe\",\"titleSlug\":\"fringe\",\"rootFolderPath\":\"/mnt/ZFSPool/TvShows/\",\"seasonFolder\":true,\"monitored\":true,\"seriesType\":\"standard\",\"qualityProfileId\":6,\"seasons\":[{\"seasonNumber\":0,\"monitored\":false},{\"seasonNumber\":1,\"monitored\":false},{\"seasonNumber\":2,\"monitored\":false},{\"seasonNumber\":3,\"monitored\":false},{\"seasonNumber\":4,\"monitored\":false},{\"seasonNumber\":5,\"monitored\":false}]}","timestamp":"2016-09-01T12:07:06.427Z"}
{"level":"info","message":"user: Dan undefined, message: added series \"Fringe\"","timestamp":"2016-09-01T12:07:06.572Z"}
{"level":"info","message":"user: Dan undefined, message: clearing series cache","timestamp":"2016-09-01T12:07:06.697Z"}
```

And this one from syslog (almost the same as predicted): 
```
Sep  1 14:06:27 dl Telegram-Sonarr[14037]: 2016-09-01T12:06:27.650Z - #033[32minfo#033[39m: user: Dan undefined, message: sent '/query' command
Sep  1 14:06:27 dl Telegram-Sonarr[14037]: 2016-09-01T12:06:27.808Z - #033[32minfo#033[39m: user: Dan undefined, message: requested to search for series "fringe"
Sep  1 14:06:27 dl Telegram-Sonarr[14037]: 2016-09-01T12:06:27.816Z - #033[32minfo#033[39m: #2 user: Dan undefined, message: found the following series Fringe - 2008
Sep  1 14:06:36 dl Telegram-Sonarr[14037]: 2016-09-01T12:06:36.163Z - #033[32minfo#033[39m: user: 208282568, message: confirm the series Fringe - 2008
Sep  1 14:06:36 dl Telegram-Sonarr[14037]: 2016-09-01T12:06:36.540Z - #033[32minfo#033[39m: user: Dan undefined, message: looking for existing series
Sep  1 14:06:36 dl Telegram-Sonarr[14037]: 2016-09-01T12:06:36.657Z - #033[32minfo#033[39m: user: Dan undefined, message: conform correct show: Fringe - 2008
Sep  1 14:06:45 dl Telegram-Sonarr[14037]: 2016-09-01T12:06:45.149Z - #033[32minfo#033[39m: user: 208282568, message: choose the series Yes
Sep  1 14:06:45 dl Telegram-Sonarr[14037]: 2016-09-01T12:06:45.159Z - #033[32minfo#033[39m: user: Dan undefined, message: requested to get profile list
Sep  1 14:06:45 dl Telegram-Sonarr[14037]: 2016-09-01T12:06:45.160Z - #033[32minfo#033[39m: #3 user: Dan undefined, message: found the following profiles Any,SD,HD-720p,HD-1080p,HD - All,WEB-DL 1080P,WEB-DL 720P,Bluray 1080P,Bluray 720p
Sep  1 14:06:52 dl Telegram-Sonarr[14037]: 2016-09-01T12:06:52.517Z - #033[32minfo#033[39m: user: 208282568, message: choose the profile "WEB-DL 1080P"
Sep  1 14:06:52 dl Telegram-Sonarr[14037]: 2016-09-01T12:06:52.518Z - #033[32minfo#033[39m: user: Dan undefined, message: requested to get monitor list
Sep  1 14:06:52 dl Telegram-Sonarr[14037]: 2016-09-01T12:06:52.518Z - #033[32minfo#033[39m: user: Dan undefined, message: found the following monitor types future,all,none,latest,first
Sep  1 14:06:55 dl Telegram-Sonarr[14037]: 2016-09-01T12:06:55.116Z - #033[32minfo#033[39m: user: 208282568, message: choose the type "none"
Sep  1 14:06:55 dl Telegram-Sonarr[14037]: 2016-09-01T12:06:55.117Z - #033[32minfo#033[39m: user: Dan undefined, message: requested to get series types list
Sep  1 14:06:55 dl Telegram-Sonarr[14037]: 2016-09-01T12:06:55.117Z - #033[32minfo#033[39m: user: Dan undefined, message: found the following series types standard,airs daily,anime
Sep  1 14:06:57 dl Telegram-Sonarr[14037]: 2016-09-01T12:06:57.192Z - #033[32minfo#033[39m: user: 208282568, message: choose the folder "standard"
Sep  1 14:06:58 dl Telegram-Sonarr[14037]: 2016-09-01T12:06:58.659Z - #033[32minfo#033[39m: user: Dan undefined, message: requested to get folder list
Sep  1 14:06:58 dl Telegram-Sonarr[14037]: 2016-09-01T12:06:58.660Z - #033[32minfo#033[39m: user: Dan undefined, message: found the following folders /mnt/ZFSPool/TvShows/
Sep  1 14:07:03 dl Telegram-Sonarr[14037]: 2016-09-01T12:07:03.296Z - #033[32minfo#033[39m: user: 208282568, message: choose the season folder "/mnt/ZFSPool/TvShows/"
Sep  1 14:07:03 dl Telegram-Sonarr[14037]: 2016-09-01T12:07:03.296Z - #033[32minfo#033[39m: user: Dan undefined, message: requested to get season folders list
Sep  1 14:07:03 dl Telegram-Sonarr[14037]: 2016-09-01T12:07:03.297Z - #033[32minfo#033[39m: user: Dan undefined, message: found the following seasons folder types yes,no
Sep  1 14:07:06 dl Telegram-Sonarr[14037]: 2016-09-01T12:07:06.426Z - #033[32minfo#033[39m: user: Dan undefined, message: adding series "Fringe" with options {"tvdbId":82066,"title":"Fringe","titleSlug":"fringe","rootFolderPath":"/mnt/ZFSPool/TvShows/","seasonFolder":true,"monitored":true,"seriesType":"standard","qualityProfileId":6,"seasons":[{"seasonNumber":0,"monitored":false},{"seasonNumber":1,"monitored":false},{"seasonNumber":2,"monitored":false},{"seasonNumber":3,"monitored":false},{"seasonNumber":4,"monitored":false},{"seasonNumber":5,"monitored":false}]}
Sep  1 14:07:06 dl mono[32688]: [Info] SeriesService: Adding Series [82066][Fringe] Path: [/mnt/ZFSPool/TvShows/Fringe]
Sep  1 14:07:06 dl Telegram-Sonarr[14037]: 2016-09-01T12:07:06.572Z - #033[32minfo#033[39m: user: Dan undefined, message: added series "Fringe"
Sep  1 14:07:06 dl Telegram-Sonarr[14037]: 2016-09-01T12:07:06.697Z - #033[32minfo#033[39m: user: Dan undefined, message: clearing series cache
```

After that, nothing shows up in sonarr for this show.

Thanks

**Comment by onedr0p** [comment]
It looks like it wasn't finding the config.json. Can you verify its in that path as described in the last part?

**Comment by danama** [comment]
Yes, the path is correct.

I have even tried to pass a 777 on everything to be sure it's not permission related

```
root@dl:/opt/telegram-sonarr-bot/lib# ll
total 28
drwxrwxrwx 2 telegram telegram 4096 sep  1 15:11 ./
drwxrwxrwx 8 telegram telegram 4096 sep  1 15:17 ../
-rwxrwxrwx 1 telegram telegram  712 aoû 29 13:06 acl.js*
-rwxrwxrwx 1 telegram telegram 1749 aoû 29 13:06 config.js*
-rwxrwxrwx 1 telegram telegram  174 aoû 29 13:06 lang.js*
-rwxrwxrwx 1 telegram telegram  337 aoû 29 13:06 logger.js*
-rwxrwxrwx 1 telegram telegram  521 aoû 29 13:06 state.js*
```

**Comment by onedr0p** [comment]
Note it is not finding `config.json` which is in the projects root folder, I think you are confused by the `config.js` in the lib folder.

**Comment by danama** [comment]
Everything is working now since the fix you have made in #54 

Thank you !

**Comment by danama** [comment]
Well.. the notifications are still the same like described in the first post but the main issue is solved
